pytorch_pretrained_bert/tokenization.py

```diff
@@ -74,25 +74,36 @@ def whitespace_tokenize(text):
 class BertTokenizer(object):
     """Runs end-to-end tokenization: punctuation splitting + wordpiece"""
 
-    def __init__(self, vocab_file, do_lower_case=True, max_len=None,
+    def __init__(self, vocab_file, do_lower_case=True, max_len=None, do_basic_tokenize=True,
                  never_split=("[UNK]", "[SEP]", "[PAD]", "[CLS]", "[MASK]")):
+        """Constructs a BertTokenizer.
+
+        Args:
+          do_lower_case: Whether to lower case the input.
```

Comment:
Could you:
- indicate that `do_lower_case` and `never_split` only have an effect when `do_basic_tokenize` is `True`?
- add a docstring for `max_len` and `vocab_file`?

Reply:
:+1: see most recent commit. Let me know if you'd like different wording/etc.

Reply:
Hi, I'm trying to use `never_split` with the multilingual model.

This one confuses me:
https://github.com/huggingface/pytorch-pretrained-BERT/blob/f3e5404880902a1bdfed2b1d47d10a6c672dc430/pytorch_pretrained_bert/tokenization.py#L91

Is this meant to be equivalent to your `do_basic_tokenize` solution?
There is no `do_wordpiece_only` anywhere to be found. 🤷‍♂️

Reply:
Ah, this is a mistake on my part. An earlier draft had a "do_wordpiece_only" flag, instead of the "do_basic_tokenize" flag; I decided the latter was clearer as to what was being included/excluded.

"do_wordpiece_only=False" would be equivalent to "do_basic_tokenize=True"

the never_split list only affects the basic tokenization.

Reply:
Thank you for clearing things up so quickly. 👍